vm/src/stdlib/time.rs

```diff
@@ -207,8 +207,22 @@ mod time {
 
     #[pyfunction]
     fn strftime(format: PyStrRef, t: OptionalArg<PyStructTime>, vm: &VirtualMachine) -> PyResult {
+        use std::fmt::Write;
+
         let instant = t.naive_or_local(vm)?;
-        let formatted_time = instant.format(format.as_str()).to_string();
+
+        let mut formatted_time = String::new();
+
+        /*
+         * chrono doesn't support all formats and it
+         * raised an error if unsupported format is supplied.
+         * If error happens, we set result as input arg
+         */
+        let result = write!(&mut formatted_time, "{}", instant.format(format.as_str()));
+        match result {
+            Ok(r) => r,
+            Err(_) => formatted_time = format.to_string(),
+        };
```

Comment:
Does this work?

```suggestion
        let formatted_time = String::try_from(instant.format(format.as_str())).unwrap_or(|| format.to_string());
```

Reply:
I am getting following error

`the trait `std::convert::From<DelayedFormat<StrftimeItems<'_>>>` is not implemented for `std::string::String``

Reply:
But I do see what you are trying to do here. I will give another pass at this.

Reply:
I have refactored code, it looks lot better now